### PR TITLE
Introduce StylingPicker component

### PR DESCRIPTION
**Pull request by scavnickyj**
JIRA: TNT-923

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-kit/src/StylingPicker/StylingPicker.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+                            onClick={handleApply}
+                        />
+                    </div>
+                )}
+            </div>
+        </div>
+    );
+};
+
+const StylingPickerWithIntl = injectIntl(StylingPickerCore);
```
Comment (kandl) [inline libs/sdk-ui-kit/src/StylingPicker/StylingPicker.tsx:183]: Nitpick: you can replace this with `useIntl()` hook

File: libs/sdk-ui-kit/src/StylingPicker/StylingPicker.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+                            {titleTooltip}
+                        </Bubble>
+                    )}
+                </BubbleHoverTrigger>
+            </div>
+            <div className="gd-styling-picker-content">
+                {isLoading ? (
+                    <DialogListLoading className="gd-styling-picker-list-loading s-styling-picker-list-loading" />
+                ) : (
+                    <>
```
Comment (kandl) [inline libs/sdk-ui-kit/src/StylingPicker/StylingPicker.tsx:108]: Optional: split this to multiple subcomponents to make it more readable (eg StylingPickerHeader / StylingPickerBody / StylingPickerFooter)

File: libs/sdk-ui-kit/styles/scss/stylingPicker.scss [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    background-color: $gd-color-white;
+    border: 1px solid $is-focused-background;
+
+    .gd-styling-picker-title {
+        display: flex;
+        padding-bottom: 10px;
+        line-height: 23px;
+        cursor: default;
+
+        h3 {
```
Comment (kandl) [inline libs/sdk-ui-kit/styles/scss/stylingPicker.scss:19]: I'm not fan of these nested css selectors (+ vague selectors as h3), imo, best is to keep everything as flat as possible (once you start moving it later, or making css overrides, it's a pain)

File: libs/sdk-ui-kit/src/StylingPicker/StylingPicker.tsx [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+const TOOLTIP_ALIGN_POINTS: IAlignPoint[] = [{ align: "bl tl", offset: { x: 10, y: 4 } }];
+
+/**
+ * @internal
+ */
+export interface IStylingPickerProps {
+    title: string;
+    basicItem: StylingPickerItem;
+    customItems: StylingPickerItem[];
+    emptyMessageElement: JSX.Element;
```
Comment (kandl) [inline libs/sdk-ui-kit/src/StylingPicker/StylingPicker.tsx:28]: `emptyMessageElement` name is a bit confusing, what about calling it just `emptyMessage` and type it as a normal jsx component? `() => JSX.Element`

**Comment by scavnickyj** [comment]
extended test

File: libs/sdk-ui-kit/src/StylingPicker/StylingPicker.tsx [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+import { IAlignPoint } from "../typings/positioning";
+
+const TOOLTIP_ALIGN_POINTS: IAlignPoint[] = [{ align: "bl tl", offset: { x: 10, y: 4 } }];
+
+/**
+ * @internal
+ */
+export interface IStylingPickerProps {
+    title: string;
+    basicItem: StylingPickerItem;
```
Comment (kandl) [inline libs/sdk-ui-kit/src/StylingPicker/StylingPicker.tsx:26]: should this be named defaultItem?
Or what about including all the items in `items` prop and then provide check for `isDefaultItem: (StylingPickerItem) => boolean`

Comment (kandl) [inline libs/sdk-ui-kit/src/StylingPicker/StylingPicker.tsx:26]: should this be named `defaultItem`?
Or what about including all the items in `items` prop and then provide check for `isDefaultItem: (StylingPickerItem) => boolean`

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1750/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9032/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1750/ : SUCCESS in 27m 20s

File: libs/sdk-ui-kit/src/StylingPicker/StylingPicker.tsx (hunk 5 above)
Comment (scavnickyj) [reply libs/sdk-ui-kit/src/StylingPicker/StylingPicker.tsx:26]: I will rename it but I would prefer to not mix these together

File: libs/sdk-ui-kit/styles/scss/stylingPicker.scss (hunk 3 above)
Comment (scavnickyj) [reply libs/sdk-ui-kit/styles/scss/stylingPicker.scss]: I flattened the styles and removed the h3

**Review by kandl — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2805

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2574/ : SUCCESS in 2m 58s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2873/ : SUCCESS in 40s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1704/ : SUCCESS in 18m 01s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10876/ : SUCCESS in 19m 29s

**Pull request merged**